ui/appui-react/src/appui-react/content/FloatingViewportContent.tsx

```diff
@@ -39,11 +41,17 @@ export interface FloatingViewportContentProps {
  */
 // istanbul ignore next
 export function FloatingViewportContent(props: FloatingViewportContentProps) { // eslint-disable-line @typescript-eslint/naming-convention
-  const { contentId, initialViewState } = props;
+  const { contentId, initialViewState, viewportRef } = props;
   const [viewport, setViewport] = React.useState<ScreenViewport | undefined>();
   const contentControl = React.useRef<FloatingViewportContentControl | undefined>();
 
-  const onViewportRef = React.useCallback((v: ScreenViewport) => setViewport(v), []);
+  const viewState = React.useMemo(() => typeof initialViewState === "function" ? initialViewState() : initialViewState, [initialViewState]);
+  const onViewportRef = React.useCallback((v: ScreenViewport) => {
+    setViewport(v);
+    if (viewportRef)
```

Comment:
Please add a test for this new branch.

Reply:
I just saw that the whole component is marked as istanbul ignore, so we'll need to jump in and write test, don't start this from scratch.